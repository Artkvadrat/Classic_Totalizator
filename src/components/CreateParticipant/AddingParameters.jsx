import React, { useState } from 'react';
import { Button, Form, Input } from 'antd';
import PropTypes from 'prop-types';

const AddingParameters = ({ initialData, id, addData, type }) => {
  const [data, setData] = useState(initialData);

  const changeHandler = (e) => {
    setData(() => ({
      ...data,
      [e.target.name]: e.target.value
    }));
  };

  const submitParameter = () => {
    addData(data, type, id);
  };

  const isDataChanged =
    initialData.type === data.type && initialData.value === data.value;

  return (
    <Form.Item layout="horizontal">
      <div style={{ display: 'flex' }}>
        <Input
          name="type"
          placeholder="Type"
          defaultValue={data.type}
          onChange={changeHandler}
          required
        />
        <Input
          name="value"
          placeholder="Value"
          defaultValue={data.value}
          onChange={changeHandler}
          required
          style={{ margin: '0 8px' }}
        />
        <Button
          type="primary"
          htmlType="button"
          disabled={!(!!data.type && !!data.value) || isDataChanged}
          onClick={submitParameter}
        >
          {initialData.type && initialData.value ? 'Save' : 'Add'}
        </Button>
      </div>
    </Form.Item>
  );
};

AddingParameters.propTypes = {
  initialData: PropTypes.object,
  id: PropTypes.number,
  addData: PropTypes.func,
  type: PropTypes.string
};

export default AddingParameters;
