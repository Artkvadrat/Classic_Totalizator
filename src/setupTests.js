import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';

enzyme.configure({ adapter: new Adapter() });

const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  clear: jest.fn()
};
window.localStorage = localStorageMock;
