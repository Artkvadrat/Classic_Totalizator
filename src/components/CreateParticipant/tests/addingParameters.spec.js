import React from 'react';
import { shallow } from 'enzyme';
import { Input, Button } from 'antd';

import '../../../setupTests';

import AddingParameters from '../AddingParameters';

describe('AddingParameters component', () => {
  let wrapper;

  beforeAll(() => {
    window.matchMedia =
      window.matchMedia ||
      function () {
        return {
          matches: false,
          addListener: () => {},
          removeListener: () => {}
        };
      };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render without error', () => {
    const props = {
      initialData: { type: 'type', value: 'value' },
      id: 0,
      addData: jest.fn(),
      type: 'player'
    };

    wrapper = shallow(<AddingParameters {...props} />);
    expect(wrapper).toMatchSnapshot();

    const firstInput = wrapper.find(Input).at(0);
    expect(firstInput.prop('defaultValue')).toEqual('type');

    const secondInput = wrapper.find(Input).at(1);
    expect(secondInput.prop('defaultValue')).toEqual('value');
  });

  it('should handle submit Form.Item', () => {
    const props = {
      initialData: { type: 'type', value: 'value' },
      id: 0,
      addData: jest.fn(),
      type: 'player'
    };

    wrapper = shallow(<AddingParameters {...props} />);

    const button = wrapper.find(Button);
    button.simulate('click');
    expect(props.addData).toHaveBeenCalled();
    expect(props.addData).toHaveBeenCalledTimes(1);
  });
});
