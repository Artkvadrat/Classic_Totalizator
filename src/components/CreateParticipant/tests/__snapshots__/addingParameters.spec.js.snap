// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddingParameters component should render without error 1`] = `
<FormItem
  layout="horizontal"
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <Input
      defaultValue="type"
      name="type"
      onChange={[Function]}
      placeholder="Type"
      required={true}
      type="text"
    />
    <Input
      defaultValue="value"
      name="value"
      onChange={[Function]}
      placeholder="Value"
      required={true}
      style={
        Object {
          "margin": "0 8px",
        }
      }
      type="text"
    />
    <Button
      disabled={true}
      htmlType="button"
      onClick={[Function]}
      type="primary"
    >
      Save
    </Button>
  </div>
</FormItem>
`;
